GetEligibilityCheck/handler.ts

```diff
@@ -0,0 +1,93 @@
+import { Context } from "@azure/functions";
+import * as df from "durable-functions";
+import * as express from "express";
+import { fromNullable } from "fp-ts/lib/Option";
+import { ContextMiddleware } from "io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import { FiscalCodeMiddleware } from "io-functions-commons/dist/src/utils/middlewares/fiscalcode";
+import {
+  withRequestMiddlewares,
+  wrapRequestHandler
+} from "io-functions-commons/dist/src/utils/request_middleware";
+import { readableReport } from "italia-ts-commons/lib/reporters";
+import {
+  IResponseErrorInternal,
+  IResponseSuccessAccepted,
+  IResponseSuccessJson,
+  ResponseErrorInternal,
+  ResponseSuccessAccepted,
+  ResponseSuccessJson
+} from "italia-ts-commons/lib/responses";
+import { FiscalCode } from "italia-ts-commons/lib/strings";
+import { ActivityResultSuccess } from "../EligibilityCheckActivity/handler";
+import { SottoSogliaEnum } from "../generated/definitions/ConsultazioneSogliaIndicatoreResponse";
+import { EligibilityCheck } from "../generated/definitions/EligibilityCheck";
+import { EligibilityCheckStatusEnum } from "../generated/definitions/EligibilityCheckStatus";
+import { initTelemetryClient } from "../utils/appinsights";
+
+type IGetEligibilityCheckHandler = (
+  context: Context,
+  fiscalCode: FiscalCode
+) => Promise<
+  | IResponseSuccessJson<EligibilityCheck>
+  | IResponseErrorInternal
+  | IResponseSuccessAccepted
+>;
+
+initTelemetryClient();
+
+// tslint:disable-next-line: cognitive-complexity
+export function GetEligibilityCheckHandler(): IGetEligibilityCheckHandler {
+  return async (context, fiscalCode) => {
+    const client = df.getClient(context);
+    const status = await client.getStatus(fiscalCode);
+    return ActivityResultSuccess.decode(status.customStatus)
+      .map(_ => {
+        const bonusValue = fromNullable(_.data.Componente)
+          .map(_1 =>
+            _1.length > 2
+              ? 50000
+              : _1.length === 2
+              ? 25000
+              : _1.length === 1
+              ? 15000
+              : 0
+          )
+          .getOrElse(0);
+        return EligibilityCheck.encode({
+          family_members: _.data.Componente || [],
+          max_amount: bonusValue,
+          max_tax_benefit: bonusValue, // TODO: Valorize properly this value
+          status:
+            _.data.SottoSoglia === SottoSogliaEnum.SI
+              ? EligibilityCheckStatusEnum.ELIGIBILE
+              : EligibilityCheckStatusEnum.INELIGIBLE
+        });
+      })
+      .fold<
+        | IResponseErrorInternal
+        | IResponseSuccessAccepted
+        | IResponseSuccessJson<EligibilityCheck>
+      >(
+        _ => {
+          if (status.runtimeStatus === df.OrchestrationRuntimeStatus.Running) {
```

Comment:
can we put this check at the start of the method?

Reply:
fixed in 0c41302